### ISSUE TITLE
[Breakdown Eng] Site permissions

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/fenix#325]
Meta: https://github.com/mozilla-mobile/fenix/issues/218
UX mocks, ready for eng: https://github.com/mozilla-mobile/fenix/issues/188#issuecomment-459217059

**Comment by cpeterso** [comment]
GV API for site permissions:
https://bugzilla.mozilla.org/show_bug.cgi?id=1524489

TBD: Should Gecko remember the site permissions (and expose a GV API for the app to get and set permissions) or should GV always ask the app (and the app will remember the permissions)? Having the app remember the permissions might be easiest: the GV API would be simpler and the app is responsible for the permission prompt and management UI anyways.

**Comment by pocmo** [comment]
The AC team is tracking work in the following project board:
https://github.com/orgs/mozilla-mobile/projects/8

We are going to provide a component that deals with handling content permission requests by showing a prompt UI (following UX mocks or having the customization hooks to make it so) or auto-accepting them based on previous user decisions. In addition to that we are going to provide the APIs needed to create settings screens based on the saved choices.

**Comment by cpeterso** [comment]
btw, GV site permissions API [bug 1524489]( https://bugzilla.mozilla.org/show_bug.cgi?id=1524489) has been closed as WONTFIX. Randall thinks GV's existing Permissions Delegate API should be adequate, so Fenix should not be blocked on GV for this feature.

**Comment by vesta0** [comment]
@pocmo is this breakdown blocked on A-C?

**Comment by bbinto** [comment]
@Amejia481 to look into breaking this down.

**Comment by Amejia481** [comment]
👍

**Comment by boek** [comment]
After discussion we feel this is too large to breakdown. We need to split this up into multiple deliverables.

**Comment by bbinto** [comment]
@mheubusch, we might want to address this in our user story session meeting tomorrow. I'm thinking of the minimal version for MVP.

**Comment by vesta0** [comment]
User stories are attached to meta #218 and Engineering tasks should be attached to each user story. As discussed with team, these Breakdown issues will be closed.
